blockchain/fastsync.go

```diff
@@ -212,34 +213,32 @@ func worker(cache *utxoCache, jobs <-chan []byte, results chan<- *result) {
 // downloadUtxoSet will attempt to connect to make an HTTP GET request to the
 // provided sources one at a time and download the UTXO set to the provided path.
 // If a proxy is provided it will use it for the HTTP connection.
-func downloadUtxoSet(sources []string, proxy *socks.Proxy, pth string) error {
-	dialFunc := net.Dial
-	if proxy != nil {
-		dialFunc = proxy.Dial
-	}
-	tr := &http.Transport{
-		Dial: dialFunc,
-	}
-	client := &http.Client{Transport: tr}
+func downloadUtxoSet(sources []string, proxy *socks.Proxy, pth string) (string, error) {
+	var fileName string
+
 	for _, src := range sources {
-		resp, err := client.Get(src)
-		if err != nil {
-			continue
-		}
-		if resp.StatusCode != http.StatusOK {
-			continue
-		}
-		file, err := os.Create(pth)
-		if err != nil {
-			return err
-		}
 		log.Infof("Downloading UTXO set from %s", src)
-		_, err = io.Copy(file, resp.Body)
-		if err != nil {
-			return err
+		retry.Do(
```

Comment:
I think we should log the error if each source fails. It might turn out that there is a more general reason for the error and the reason will be invisible in the logs except that somehow they all failed.

```go
		err := retry.Do(
		// ...
		if err != nil {
			log.Warnf("Unable to download UTXO set from %s: %v", src, err)
		}
```

Reply:
We can add error wrapping for sure. However that would only bubble up if all failed. Or we can print when failures occur and the retry would kick in.

Reply:
Now it logs all errors, and gives retry stats. =)